src/name_bazaar/ui/core.cljs

```diff
@@ -47,9 +47,11 @@
                    {:async-flow {:first-dispatch [:district0x/load-smart-contracts {:version constants/contracts-version}]
                                  :rules [{:when :seen?
                                           :events [:district0x/smart-contracts-loaded :district0x/my-addresses-loaded]
-                                          :dispatch-n [[:district0x/watch-my-eth-balances]
-                                                       [:active-page-changed]
-                                                       [:try-resolving-address]]}]}
+                                          :dispatch-n (remove nil?
+                                                              [[:district0x/watch-my-eth-balances]
+                                                               (when history/hashroutes?
+                                                                 [:try-resolving-address]
+                                                                 [:active-page-changed])])}]}
```

Comment:
this will return only `[:active-page-changed]`. Must write it as 2 when statements